interface-definitions/service_ids_suricata.xml.in

```diff
@@ -0,0 +1,136 @@
+<?xml version="1.0"?>
+<interfaceDefinition>
+  <node name="service">
+    <children>
+      <node name="ids">
+        <properties>
+          <help>Intrusion Detection System</help>
+        </properties>
+        <children>
+          <node name="suricata" owner="${vyos_conf_scripts_dir}/service_ids_suricata.py">
+            <properties>
+              <help>Suricata high performance Network IDS, IPS and Network Security Monitoring engine</help>
+              <priority>740</priority>
+            </properties>
+            <children>
+              #include <include/generic-interface-multi.xml.i>
+              <tagNode name="address-group">
+                <properties>
+                  <help>Address group name</help>
+                  <completionHelp>
+                    <list>home-net external-net http-servers smtp-servers sql-servers dns-servers telnet-servers aim-servers dc-servers dnp3-server dnp3-client modbus-client modbus-server enip-client enip-server</list>
+                  </completionHelp>
+                  <constraint>
+                    <regex>[a-z0-9-]+</regex>
+                  </constraint>
+                </properties>
+                <children>
+                  <leafNode name="network">
+                    <properties>
+                      <help>Address group values</help>
+                      <valueHelp>
+                        <format>txt</format>
+                        <description>Named address group (e.g., $home-net)</description>
+                      </valueHelp>
```

Comment:
1. If it can include another `address-group` we probably should add the completion help
```
              <completionHelp>
                <path>service ids suricata address-group</path>
              </completionHelp>
```

2. Do we need to check if we refer to the non-existing address group?
```
vyos@r4# set service ids suricata address-group home-net network 
Possible completions:
   <text>               Named address group (e.g., $home-net)
   <x.x.x.x/x>          IPv4 prefix(es) to group
   <h:h:h:h:h:h:h:h/x>  IPv6 prefix(es) to group
   100.64.0.0/24        
   100.64.1.0/28        

      
[edit]
vyos@r4# set service ids suricata address-group home-net network foo
[edit]
vyos@r4# commit
[ service ids suricata ]

WARNING: To fetch the latest rules, use "update suricata"; To
periodically fetch the latest rules, use the task scheduler!

Job for suricata.service failed because the control process exited with error code.
See "systemctl status suricata.service" and "journalctl -xeu suricata.service" for details.

[edit]
vyos@r4#
```

Reply:
I'll add support (after a quick rebase) to reference other groups using the `completionHelp` and `$`-prefix them in the template. I'll also add group validation.

Reply:
You probably do not need to rebase as it is a new feature
It is OK if something else is changed in other files (not related to this PR)

Reply:
What do you think about `include` usage like in the firewall
```
vyos@r4# set firewall group address-group foo address 192.0.2.1
[edit]
vyos@r4# set firewall group address-group bar address 192.0.2.5
[edit]
vyos@r4# set firewall group address-group bar include foo 
[edit]
vyos@r4# 
```

Reply:
Rebase was in a hope to fix my broken ISO generation due to `podman` requiring `libgpgme11t64 (>= 1.4.1)`; I'll address the above changes in the coming days.

Thanks for the constructive and extensive feedback!

Reply:
> What do you think about `include` usage like in the firewall

We could indeed opt to use `include` and `exclude` followed by the type `address-group`, `network` or `address`to get as close as possible to the [source/destination examples](https://docs.suricata.io/en/latest/rules/intro.html#source-and-destination).

```
# HOME_NET = [192.168.0.0/16,127.0.0.0/8,!192.168.1.0/24]
set address-group home-net include network 192.168.0.0/16
set address-group home-net include network 127.0.0.0/8
set address-group home-net exclude network 192.168.1.0/24

# EXTERNAL_NET = [!$HOME_NET]
set address-group external-net exclude address-group home-net
```

```
# HTTP_SERVERS = [192.168.3.0/24]
set address-group http-servers include network 192.168.3.0/24

# SQL_SERVERS = [192.168.4.0/24]
set address-group sql-servers include network 192.168.4.0/24

# SQL_BACKUPS = [192.168.4.4]
set address-group sql-backups include address 192.168.4.4

# DMZ = [$HTTP_SERVERS,$SQL_SERVERS,!$SQL_BACKUPS]
set address-group dmz include address-group http-servers
set address-group dmz include address-group sql-servers
set address-group dmz exclude address-group sql-backups
```

The same would apply to the `port-group`. I just have to figure out some logic to error on circular dependencies and define the order of placement inside the configuration.

Reply:
I opted to align the usage with other commands instead:
```
# HOME_NET = [192.168.0.0/16,127.0.0.0/8,!192.168.1.0/24]
set address-group home-net address 192.168.0.0/16
set address-group home-net address 127.0.0.0/8
set address-group home-net address !192.168.1.0/24

# EXTERNAL_NET = [!$HOME_NET]
set address-group external-net group home-net
```
```
# HTTP_SERVERS = [192.168.3.0/24]
set address-group http-servers address 192.168.3.0/24

# SQL_SERVERS = [192.168.4.0/24]
set address-group sql-servers address 192.168.4.0/24

# SQL_BACKUPS = [192.168.4.4]
set address-group sql-backups address 192.168.4.4

# DMZ = [$HTTP_SERVERS,$SQL_SERVERS,!$SQL_BACKUPS]
set address-group dmz group http-servers
set address-group dmz group sql-servers
set address-group dmz group !sql-backups
```